Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -3596,25 +3596,21 @@ void WebPageProxy::sendKeyEvent(const NativeWebKeyboardEvent& event)
     m_process->recordUserGestureAuthorizationToken(event.authorizationToken());
 
     auto handleKeyEventReply = [this, protectedThis = Ref { *this }] (std::optional<WebEventType> eventType, bool handled) {
-        if (!m_pageClient)
-            return;
-        if (!eventType)
+        if (!m_pageClient || !eventType) {
+            internals().keyEventQueue.removeFirst();
```

Comment:
Should this still `MESSAGE_CHECK` if `internals().mouseEventQueue.isEmpty()`?

The implementation of `removeFirst()` looks like it would otherwise crash the UI process in the case where it's empty:

```
template<typename T, size_t inlineCapacity>
inline void Deque<T, inlineCapacity>::removeFirst()
{
    checkValidity();
    invalidateIterators();
    RELEASE_ASSERT(!isEmpty());
```

Reply:
Yeah we should probably have the same `MESSAGE_CHECK` here as well. Nice catch, will fix.